src/PIL/ImageFont.py

```diff
@@ -185,7 +188,14 @@ def getlength(self, text, *args, **kwargs):
 class FreeTypeFont:
     """FreeType font wrapper (requires _imagingft service)"""
 
-    def __init__(self, font=None, size=10, index=0, encoding="", layout_engine=None):
+    def __init__(
+        self,
+        font: bytes | str | Path | None = None,
+        size: float = 10,
+        index: int = 0,
+        encoding: str = "",
+        layout_engine: int | None = None,
```

Comment:
Testing `Layout | None`, I find that `layout_engine=ImageFont.Layout.RAQM` is fine, but `layout_engine=1` gives a warning in PyCharm.
![pycharm](https://github.com/python-pillow/Pillow/assets/3112309/fcf6afc7-3922-4732-9f5e-83db1eb17a76)

Is that helpful or unhelpful?

Reply:
I think that's helpful -- if people are adding types to their code, they should use `Layout` enums rather than `int`s.

And typeshed also has `Layout | None` here:

https://github.com/python/typeshed/blob/9fbf2e1c0cecb6d4c855fa283e1f3c6030006485/stubs/Pillow/PIL/ImageFont.pyi#L90